datafusion/optimizer/src/decorrelate_predicate_subquery.rs

```diff
@@ -248,6 +248,38 @@ fn build_join(
                 .map(Option::Some)
         })?;
 
+    // build a predicate for comparing the left and right expressions of a given pair of outer/subquery rows
+    // from an IN or NOT IN predicate
+    let build_in_predicate = |left: Box<Expr>, right: Box<Expr>| -> Result<Expr> {
+        let right_col = create_col_from_scalar_expr(right.deref(), Some(subquery_alias))?;
+        let eq_predicate = Expr::eq(left.deref().clone(), Expr::Column(right_col));
+        if !query_info.negated {
+            // early exit if this is an IN predicate
+            return Ok(eq_predicate);
+        }
+
+        match left.nullable(outer_query.schema())? {
+            true => {
+                // left expression is nullable; we know the predicate must take the form `left = right IS NOT FALSE`
+                return Ok(eq_predicate.is_not_false());
+            }
+            false => {}
+        }
+        let unqualified_right_col = create_col_from_scalar_expr(right.deref(), None)?;
+        let subquery_col = query_info
+            .query
+            .subquery
+            .schema()
+            .field_from_column(&unqualified_right_col)?;
```

Comment:
Could we use  `field_with_unqualified_name` here? The unqualified name can be cloned from `right_col.name`.
So that there is no need to create `unqualified_right_col`, and we do not need to modify the `create_col_from_scalar_expr` function.

Reply:
Thanks! I've made the suggested change.